src/db.ts

```diff
@@ -320,6 +320,10 @@ export class Db {
    * @returns return the new Collection instance
    */
   collection<TSchema extends Document = Document>(name: string): Collection<TSchema>;
+  collection<TSchema extends Document = Document>(
```

Comment:
just curious why this was required

Reply:
This is a bug, when you have any overrides then you need to enumerate all the possibilities above the actual method declaration, so currently TS doesn't allow options to be passed in.

Reply:
yeah in this case I'm not sure why we want to have overrides instead of just the optional options definition... imho fewer definitions is better, but not a hill I'm willing to die on if you have strong opinions otherwise

Reply:
GH is not showing as outdated but I changed it to:
```ts
  /**
   * Returns a reference to a MongoDB Collection. If it does not exist it will be created implicitly.
   *
   * @param name - the collection name we wish to access.
   * @returns return the new Collection instance
   */
  collection<TSchema extends Document = Document>(
    name: string,
    options: CollectionOptions = {}
  ): Collection<TSchema> {
    if (typeof options === 'function') {
      // TODO(NODE-3485): Replace this with MongoDeprecationError
      throw new MongoDriverError('The callback form of this helper has been removed.');
    }
    const finalOptions = resolveOptions(this, options);
    return new Collection<TSchema>(this, name, finalOptions);
  }
 ```